lightwood/data/encoded_ds.py

```diff
@@ -35,20 +44,25 @@ def __getitem__(self, idx: int) -> Tuple[torch.Tensor, torch.Tensor]:
 
         X = torch.FloatTensor()
         for col in self.data_frame:
-            kwargs = {}
-            if 'dependency_data' in inspect.signature(self.encoders[col].encode).parameters:
-                kwargs['dependency_data'] = {dep: [self.data_frame.iloc[idx][dep]]
-                                             for dep in self.encoders[col].dependencies}
-
-            encoded_tensor = self.encoders[col].encode([self.data_frame.iloc[idx][col]], **kwargs)[0]
-            if col != self.target:
-                X = torch.cat([X, encoded_tensor])
-            else:
-                Y = encoded_tensor
+            if self.encoders.get(col, None):
+                kwargs = {}
+                if 'dependency_data' in inspect.signature(self.encoders[col].encode).parameters:
+                    kwargs['dependency_data'] = {dep: [self.data_frame.iloc[idx][dep]]
+                                                 for dep in self.encoders[col].dependencies}
+                if hasattr(self.encoders[col], 'data_window'):
+                    cols = [self.target] + [f'{self.target}_timestep_{i}' for i in range(1, self.encoders[col].data_window)]
+                    data = self.data_frame[cols].iloc[idx].tolist()
+                else:
+                    data = [self.data_frame.iloc[idx][col]]
```

Comment:
in the else
`cols = [col]`

and then bellow for both branches you can do

`data = self.data_frame[cols].iloc[idx].tolist()`